webrender/src/resource_cache.rs

```diff
@@ -445,6 +471,23 @@ impl ResourceCache {
         }
     }
 
+    pub fn request_geometry(&mut self,
```

Comment:
I see the reasoning for treating the SVG dimensions, coming from the primitive rectangle, as a part of the key. I don't think we should do that though - it's breaking the primitive abstraction a bit. Instead, I'd like to see the rasterized dimensions being passed to `update_geometry`. This way we'd have:
  1. geometry being rendering (on screen) via the existing image API, as @nical suggested
  2. better support for transformed SVG primitives